### DISCUSSION TITLE
Why does this trigger event not happen until I move off the tile?

### DISCUSSION THREAD

**Discussion by splashsky** [Annoraaq/grid-engine#158] [Q&A]
I'm in the middle of experimenting with the `movementStopped` observable, and it was going fairly swell. I took the "trigger" tile example from the `positionChanged` demo and tried to instead apply it to `movementStopped`, and it works with one caveat.

Here's the code...

```ts
// In the create() function of the scene
this.GridEngine.movementStopped().subscribe(({ charId, direction }) => {
    console.log("Movement stopped");
    
    if (this.hasTrigger(tilemap, this.GridEngine.getPosition(charId))) {
        console.log("Found the trigger!");
        notice("Found the thingy!");
    }
});

private hasTrigger(tilemap: Phaser.Tilemaps.Tilemap, pos: Position): boolean
{
    console.log("Checking for trigger");

    const result = tilemap.layers.some((layer) => {
        const tile = tilemap.getTileAt(pos.x, pos.y, false, layer.name);
        return tile ? 'trigger' in tile.properties : false;
    });

    console.log(result ? "Found it!" : "Nope.");

    return result;
}
```

And the console will print "checking for trigger" and "nope" on all the correct tiles. However, when I step onto the "trigger" tile, it still says "nope"...but then I move off the "trigger" tile and **then** it prints "found it!". Why is this? Am I checking wrong?

**Comment by Annoraaq** [comment]
I think I know the issue. The problem is, that the observable is firing before the new position is set. I guess this could also be the cause for the bug that you were describing in #159 ?

**Reply by splashsky** [reply]
That'd be correct. The suggestion of creating a new `movementChangeStopped` Observable would address this flawlessly. Thanks!
